### ISSUE TITLE
[6.7] running eslint through npm script throws error.

### ISSUE THREAD

**Issue by kufii** [eslint/eslint#12604]
I don't experience the issue when using eslint v6.6.x.

I have the following script in my package.json:

```javascript
  "scripts": {
    "lint": "eslint ."
  },
```

If I run `eslint .` directly in the console it works correctly, but when I run `npm run lint`, it throws an error. It seems like it's using the wrong config?

```
ESLint: 6.7.1.

ESLint couldn't find the config "adpyke-es6" to extend from. Please check that the name of the config is correct.

The config "adpyke-es6" was referenced from the config file in "C:\Users\pykea\.eslintrc".

If you still have problems, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

**Comment by adamklepacz** [comment]
I have the same problem but with eslint-prettier-plugin. 
I've been adding prettier to eslint on Saturday and everything works fine, but today on the new project I've got this:

```Oops! Something went wrong! :(

ESLint: 6.7.1.

ESLint couldn't find the config "prettier" to extend from. Please check that the name of the config is correct.

The config "prettier" was referenced from the config file in "/Users/adamklepacz/GitHub/weather_tracker/node_modules/eslint-plugin-prettier/eslint-plugin-prettier.js".
```

Should I downgrade the eslint to the previous release?

**Comment by Digznav** [comment]
Same here. I was updating the dependencies today and the error pop up when testing with `eslint@6.7.1`

```
$ eslint --report-unused-disable-directives .


Oops! Something went wrong! :(

ESLint: 6.7.1.

ESLint couldn't find the config "airbnb" to extend from. Please check that the name of the config is correct.

The config "airbnb" was referenced from the config file in "/Users/diegomiguel/.eslintrc.json".
```

I then installed a previous version `eslint@6.6.0`, and it worked as expected.

**Comment by mysticatea** [comment]
Hi. Thank you for your reports.

Would you provide the repro steps from scratch?
I'm troubled to reproduce it.

Thanks.

**Comment by kufii** [comment]
```sh
npm init
npm install --save-dev eslint eslint-config-adpyke-es6
```
create `.eslintrc` in the root of the project, use these contents

```javascript
{
  "extends": "adpyke-es6"
}
```
add the following script to `package.json`

```javascript
"lint": "eslint ."
```
run `npm run lint`

**Comment by kufii** [comment]
also it might be important to have a .eslintrc in your home folder as well? so if this doesn't work try adding .eslintrc to your home folder with the same contents

**Comment by mysticatea** [comment]
Thank you.

Hmm, I put `{"extends": "adpyke-es6"}` and `eslint-config-adpyke-es6` into both cases of the current directory and home directory, but I couldn't reproduce it.

**Comment by kufii** [comment]
can I see your package.json?

**Comment by mysticatea** [comment]
```json
{
  "name": "sandbox",
  "version": "0.0.0",
  "description": "",
  "main": "index.js",
  "files": [],
  "scripts": {
    "try": "eslint ."
  },
  "keywords": [],
  "author": "Toru Nagashima <public@mysticatea.dev>",
  "license": "MIT",
  "dependencies": {
    "eslint": "^6.7.1",
    "eslint-config-adpyke-es6": "^1.4.13"
  }
}
```

**Comment by kufii** [comment]
that looks right. And you have an .eslintrc in both your project folder and your home folder?

**Comment by mysticatea** [comment]
Yes, I have tried either and both.

**Comment by Digznav** [comment]
I have an `.eslintrc.json` in my home folder:

```
{
  "extends": [
    "airbnb",
    "prettier"
  ],
  "plugins": ["prettier"],
  "parser": "babel-eslint"
}
```

And also have a `.eslintrc.json` in my project folder: `~/projects/packages/eslint-config`

In which I have the following structure:

```
eslint-config/
	- rules/
	- index.js
	- .eslintrc.json
	- package.json
```

In config file:

```
{
  "extends": "./index"
}
```

`index.js`:

```
```

**Comment by mysticatea** [comment]
Ah, OK, I have found what it happens. Thank you for providing the information.

This happens under the following conditions:

1. You have ESLint config in your HOME directory, but the config doesn't work because of lacking dependencies in your HOME directory. (this behavior has been same since 6.0.0)
1. You have ESLint config in your project directory as well and it works well.
1. You started linting with the project directory itself (i.e., `eslint .`).

In the situation, first, ESLint has checked if the directory is ignored or not. Previously this process loads only `.eslintignore`. But since 6.7.0, this process additionally loads config files from the directory that the project directory exists because it uses `ignorePatterns` property of config files. If config files were not found in the directory that the project directory exists, it loads the config file in the HOME directory.

I think it's not intuitive if `eslint .` command loads config files outside of the project directory.

**Comment by mysticatea** [comment]
**Side note:** As you may know, [we have deprecated the config files in HOME directory](https://eslint.org/blog/2019/11/eslint-v6.7.0-released). The error of this issue looks like thrown from unused config files in HOME directory (because it should not work since ESLint 6.0.0), therefore I'd like to recommend to remove it.

**Comment by bertho-zero** [comment]
You can replace
```json
  "scripts": {
    "lint": "eslint ."
  }
```

with
```json
  "scripts": {
    "lint": "eslint $(pwd)/."
  }
```
